src/proposals/20250403/SparkEthereum_20250403.t.sol

```diff
@@ -0,0 +1,536 @@
+        controller.redeemSuperstate(ustbShares / 100);
+
+        assertApproxEqAbs(usdc.balanceOf(address(ctx.proxy)), mintAmount * 1/100, 100);
+        assertApproxEqAbs(ustb.balanceOf(address(ctx.proxy)), ustbShares * 99/100, 1);
+
+        // You can always burn the whole amount by doing it offchain
+        uint256 ustbBalance = ustb.balanceOf(address(ctx.proxy));
+        vm.prank(ctx.relayer);
+        controller.transferAsset(address(ustb), address(ustb), ustbBalance);
+
+        // USDC will come back async
+        assertApproxEqAbs(usdc.balanceOf(address(ctx.proxy)), mintAmount * 1/100, 100);
+        assertEq(ustb.balanceOf(address(ctx.proxy)), 0);
+    }
+
+    function test_ETHEREUM_centrifugeJTRSYOnboarding() public onChain(ChainIdUtils.Ethereum()) {
+        SparkLiquidityLayerContext memory ctx = _getSparkLiquidityLayerContext();
+        
+        MainnetController controller = MainnetController(ETHEREUM_NEW_ALM_CONTROLLER);
+
+        bytes32 depositKey = RateLimitHelpers.makeAssetKey(
+            controller.LIMIT_7540_DEPOSIT(),
+            CENTRIFUGE_JTRSY_VAULT
+        );
+        bytes32 withdrawKey = RateLimitHelpers.makeAssetKey(
+            controller.LIMIT_7540_REDEEM(),
+            CENTRIFUGE_JTRSY_VAULT
+        );
+
+        _assertRateLimit(depositKey,  0, 0);
+        _assertRateLimit(withdrawKey, 0, 0);
+
+        executeAllPayloadsAndBridges();
+
+        _assertRateLimit(depositKey, 200_000_000e6, 100_000_000e6 / uint256(1 days));
+        _assertRateLimit(withdrawKey, type(uint256).max, 0);
+
+        IERC20 usdc  = IERC20(Ethereum.USDC);
+        IERC20 jtrsy = IERC20(CENTRIFUGE_JTRSY_TOKEN);
+
+        // USDS -> USDC limits are 200m, go a bit below in case some is in use
+        uint256 mintAmount = 190_000_000e6;
+        vm.startPrank(ctx.relayer);
+        controller.mintUSDS(mintAmount * 1e12);
+        controller.swapUSDSToUSDC(mintAmount);
+
+        assertEq(usdc.balanceOf(address(ctx.proxy)),  mintAmount);
+        assertEq(jtrsy.balanceOf(address(ctx.proxy)), 0);
+
+        assertEq(ctx.rateLimits.getCurrentRateLimit(depositKey), 200_000_000e6);
+
+        controller.requestDepositERC7540(CENTRIFUGE_JTRSY_VAULT, mintAmount);
+        vm.stopPrank();
+
+        assertEq(ctx.rateLimits.getCurrentRateLimit(depositKey), 200_000_000e6 - mintAmount);
+
+        assertEq(usdc.balanceOf(address(ctx.proxy)),  0);
+        assertEq(jtrsy.balanceOf(address(ctx.proxy)), 0);
+
+        _centrifugeFulfillDepositRequest(mintAmount);
+
+        assertEq(usdc.balanceOf(address(ctx.proxy)),  0);
+        assertEq(jtrsy.balanceOf(address(ctx.proxy)), 0);
+
+        vm.prank(ctx.relayer);
+        controller.claimDepositERC7540(CENTRIFUGE_JTRSY_VAULT);
+
+        assertEq(usdc.balanceOf(address(ctx.proxy)),  0);
+        assertEq(jtrsy.balanceOf(address(ctx.proxy)), mintAmount / 2);
+
+        vm.prank(ctx.relayer);
+        controller.requestRedeemERC7540(CENTRIFUGE_JTRSY_VAULT, mintAmount / 2);
```

Comment:
```solidity
        assertEq(ctx.rateLimits.getCurrentRateLimit(withdrawKey), type(uint256).max);

        vm.prank(ctx.relayer);
        controller.requestRedeemERC7540(CENTRIFUGE_JTRSY_VAULT, mintAmount / 2);

        assertEq(ctx.rateLimits.getCurrentRateLimit(withdrawKey), type(uint256).max);
```

Reply:
We test this already at the top that the rate limit is unlimited. I don't think this provides anything else and is more of a test that unlimited rate limits work correctly which is unnecessary imo.